File: Java-Core/src/main/java/presto/literal/DecimalLiteral.java
package presto.literal;

import presto.runtime.Context;
import presto.type.DecimalType;
import presto.type.IType;
import presto.value.Decimal;

public class DecimalLiteral extends Literal<Decimal> {

	public DecimalLiteral(String text) {
		super(text,Decimal.Parse(text));
	}

	@Override
	public IType check(Context context) {
		return DecimalType.instance();
	}
}

File: Java-CodeServer/src/main/resources/js/prompto/literal/DecimalLiteral.js
var Literal = require("./Literal").Literal;
var Decimal = require("../value/Decimal").Decimal;
var DecimalType = require("../type/DecimalType").DecimalType;

function DecimalLiteral(text) {
	Literal.call(this, text, Decimal.Parse(text));
	return this;
}

DecimalLiteral.prototype = Object.create(Literal.prototype);
DecimalLiteral.prototype.constructor = DecimalLiteral;

DecimalLiteral.prototype.check = function(context) {
	return DecimalType.instance;
};

exports.DecimalLiteral = DecimalLiteral;